samcli/lib/build/bundler.py

```diff
@@ -11,12 +12,15 @@
 LOG = logging.getLogger(__name__)
 
 ESBUILD_PROPERTY = "esbuild"
+HANDLER_WITHOUT_FUNCTION_INDEX = 0
```

Comment:
I'm squeamish about magic numbers, do we know if there any enough safeguards for this?

Reply:
I can't find any exact documentation, but I do believe the patter is always `<path_to_file>.<handler_function_name>` for JavaScript Lambda functions. There's a brief mention of it in the documentation here: https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html

Reply:
I will add a comment to the constant.

Reply:
Removed all magic numbers if favour of file objects.